packages/core/upload/server/services/folder.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const uuid = require('uuid/v4');
+const { trimChars, trimCharsEnd, trimCharsStart } = require('lodash/fp');
+
+// TODO: to use once https://github.com/strapi/strapi/pull/12534 is merged
+// const { joinBy } = require('@strapi/utils');
+
+const folderModel = 'plugin::upload.folder';
+
+const joinBy = (joint, ...args) => {
+  const trim = trimChars(joint);
+  const trimEnd = trimCharsEnd(joint);
+  const trimStart = trimCharsStart(joint);
+
+  return args.reduce((url, path, index) => {
+    if (args.length === 1) return path;
+    if (index === 0) return trimEnd(path);
+    if (index === args.length - 1) return url + joint + trimStart(path);
+    return url + joint + trim(path);
+  }, '');
```

Comment:
I'm wondering whether you rebuilt `path.join()`, which could be used instead?

Reply:
Not exactly. `path.join()` would work as expected on linux systems but not on windows as it would put `\` instead of `/` in the path

Reply:
You are right. Still: would it be better to utilize `url-join` here and to normalize (trim) the path fragments before?

Reply:
Since we have a lighter already existing function for that, I would use it instead of installing a package, no?
Good question for the trim, I have a to think about, can we have spaces in the path ? It has some impact because the name "a name" would collide with "aname". Good catch

Reply:
I updated the code. Instead of trimming there, I forbid the name to start or end with a namespace. How does that sound? The frontend can trim the user input if needed.

Also, I think the path can contains whitespaces.

Reply:
Yah, that sounds good. 👍🏼 And I think you are right - I guess the path should be be URL encoded and therefore `url-join` would be the wrong choice!